code/modules/food&drinks/drinks/drinks/bottle.dm

```diff
@@ -243,3 +262,62 @@
 	item_state = "carton"
 	isGlass = 0
 	list_reagents = list("limejuice" = 100)
+
+
+////////////////////////// MOLOTOV ///////////////////////
+/obj/item/weapon/reagent_containers/food/drinks/bottle/molotov
+	name = "Molotov cocktail"
+	desc = "A throwing weapon used to ignite things, typically filled with an accelerant. Recommended highly by rioters and revolutionaries. Light and toss."
+	icon_state = "vodkabottle"
+	list_reagents = list()
+	var/list/accelerants = list(	/datum/reagent/consumable/ethanol,/datum/reagent/fuel,/datum/reagent/clf3,/datum/reagent/phlogiston,
+							/datum/reagent/napalm,/datum/reagent/hellwater,/datum/reagent/toxin/plasma,/datum/reagent/toxin/spore_burning)
+	var/active = 0
+
+/obj/item/weapon/reagent_containers/food/drinks/bottle/molotov/CheckParts()
+	var/obj/item/weapon/reagent_containers/food/drinks/bottle/B = locate() in contents
+	if(B)
+		icon_state = B.icon_state
+		B.reagents.copy_to(src,100)
+		if(!B.isGlass)
+			desc += " You're not sure if making this out of a carton was the brightest idea."
+			isGlass = 0
+	return
+
+/obj/item/weapon/reagent_containers/food/drinks/bottle/molotov/throw_impact(atom/target,mob/thrower)
+	var/firestarter = 0
+	for(var/datum/reagent/R in reagents.reagent_list)
+		for(var/A in accelerants)
+			if(istype(R,A))
+				firestarter = 1
+				break
+	SplashReagents(target)
+	if(firestarter && active)
+		target.fire_act()
+		new /obj/effect/hotspot(get_turf(target))
+	..()
+
+/obj/item/weapon/reagent_containers/food/drinks/bottle/molotov/attackby(obj/item/I, mob/user, params)
+	if(is_hot(I))
+		active = 1
+		user << "<span class='info'>You light the [src] on fire.</span>"
```

Comment:
Could also probably do with a message that stands out more, like using the warning span class. Also this will say "You light the the molotov cocktail on fire" so you need to scrap the "the".